client/src/pages/chat/Chat.js

```diff
@@ -1,9 +1,119 @@
-import React from "react";
+import React, { useState, useEffect, useContext } from "react";
+import { animateScroll } from "react-scroll";
+import { Button, Grid, TextField, Box } from "@material-ui/core";
 
+import { AppContext } from "../../App";
+import ChatIO, { useRecievedMessage } from "../../socket_io/ChatIO";
+import Dialog from "./Dialog";
 import "./Chat.css";
 
+const CHAT_LOG_ELEMENT_ID = "chat-log";
+const ROOM = "matchId_redTeam";
+
 function Chat() {
-  return <div>Chat</div>;
+  const { chatIO } = useContext(AppContext);
+  const [inputText, setInputText] = useState("");
+  const [log, setLog] = useState([]);
+  const recentMessage = useRecievedMessage(chatIO.state.io);
+
+  useEffect(() => {
+    const action = {
+      type: ChatIO.ACTION_TYPE.CONNECT,
+      payload: {
+        room: ROOM,
+      },
+    };
+    chatIO.dispatch(action);
+
+    return () => {
+      action.type = ChatIO.ACTION_TYPE.DISCONNECT;
+      chatIO.dispatch(action);
+    };
+  }, [chatIO]);
+
+  useEffect(() => {
+    // listen to messages being emitted from backend's socket
+    if (recentMessage) {
+      const newLog = [...log, recentMessage];
+      setLog(newLog);
+    }
+  }, [recentMessage]);
+
+  useEffect(() => {
+    scrollToBottom();
+  }, [log]);
+
+  const onInputTextChange = (event) => {
+    const value = event.target.value;
+    setInputText(value);
+  };
+
+  const sendMessage = (event, inputText) => {
+    event.preventDefault();
+    const message = { from: "Tony", text: inputText, room: ROOM }; // TODO: replace 'from' with name from user data and add user id
+    const action = {
+      type: ChatIO.ACTION_TYPE.SEND_MESSAGE,
+      payload: message,
+    };
+    chatIO.dispatch(action);
+    setInputText("");
+  };
+
+  const scrollToBottom = () => {
+    animateScroll.scrollToBottom({
+      containerId: CHAT_LOG_ELEMENT_ID,
+    });
+  };
+
+  const generateChatLog = () => {
+    return log.map((message, index) => {
+      return <Dialog key={index} from={message.from} text={message.text} />;
+    });
+  };
+
+  return (
+    <div id="chat">
+      <Box component="div" id={`${CHAT_LOG_ELEMENT_ID}`}>
+        {log.length > 0 ? generateChatLog() : null}
+      </Box>
+      <form
+        onSubmit={(event) => {
+          sendMessage(event, inputText);
```

Comment:
instead of sending a state into a function, since the function is part of this component, can you just access the state there?

Reply:
Great catch. I didn't realize that haha. Thanks